cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/partitioned/DynamicPartitionerWriterWrapper.java

```diff
@@ -102,7 +104,7 @@ TaskAttemptContext getKeySpecificContext(PartitionKey partitionKey) throws IOExc
    * @throws IOException
    */
   RecordWriter<K, V> getBaseRecordWriter(TaskAttemptContext job) throws IOException, InterruptedException {
-    return fileOutputFormat.getRecordWriter(job);
+    return createFileOutputFormat().getRecordWriter(job);
```

Comment:
maybe put a comment, referencing the JIRA, explaining why we can't cache/reuse the FileOutputFormat?